R/teal_transform_module.R

```diff
@@ -0,0 +1,224 @@
+#' For simple transformations without complex reactivity, the `server` function might look like this:s
+#'
+#' ```
+#' function(id, data) {
+#'   moduleServer(id, function(input, output, session) {
+#'     reactive({
+#'       within(
+#'         data(),
+#'         expr = x <- subset(x, col == level),
+#'         level = input$level
+#'       )
+#'     })
+#'   })
+#' }
+#' ```
+#'
+#' The example above can be simplified using `make_teal_transform_server`, where `level` is automatically matched to the
+#' corresponding `input` parameter:
+#'
+#' ```
+#' make_teal_transform_server(expr = expression(x <- subset(x, col == level)))
+#' ```
+#' @inheritParams teal_data_module
+#' @param server (`function(id, data)` or `expression`)
+#' A `shiny` module server function that takes `id` and `data` as arguments, where `id` is the module id and `data`
+#' is the reactive `teal_data` input. The `server` function must return a reactive expression containing a `teal_data`
+#' object. For simplified syntax, use [`make_teal_transform_server()`].
+#' @param datanames (`character`)
+#' Specifies the names of datasets relevant to the module. Only filters for the specified `datanames` will be displayed
+#' in the filter panel. The keyword `"all"` can be used to display filters for all datasets. `datanames` are
+#' automatically appended to the [`modules()`] `datanames`.
+#'
+#'
+#' @examples
+#' data_transformators <- list(
+#'   teal_transform_module(
+#'     label = "Static transformator for iris",
+#'     datanames = "iris",
+#'     server = function(id, data) {
+#'       moduleServer(id, function(input, output, session) {
+#'         reactive({
+#'           within(data(), {
+#'             iris <- head(iris, 5)
+#'           })
+#'         })
+#'       })
+#'     }
+#'   ),
+#'   teal_transform_module(
+#'     label = "Interactive transformator for iris",
+#'     datanames = "iris",
+#'     ui = function(id) {
+#'       ns <- NS(id)
+#'       tags$div(
+#'         numericInput(ns("n_cols"), "Show n columns", value = 5, min = 1, max = 5, step = 1)
+#'       )
+#'     },
+#'     server = function(id, data) {
+#'       moduleServer(id, function(input, output, session) {
+#'         reactive({
+#'           within(data(),
+#'             {
+#'               iris <- iris[, 1:n_cols]
+#'             },
+#'             n_cols = input$n_cols
+#'           )
+#'         })
+#'       })
+#'     }
+#'   )
+#' )
+#'
+#' output_decorator <- teal_transform_module(
+#'   server = make_teal_transform_server(
+#'     expression(
+#'       object <- rev(object)
+#'     )
+#'   )
+#' )
+#'
+#' app <- init(
+#'   data = teal_data(iris = iris),
+#'   modules = example_module(transformators = data_transformators, decorators = output_decorator)
+#' )
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+#'
+#' @name teal_transform_module
+#'
+#' @export
+teal_transform_module <- function(ui = NULL,
```

Comment:
what is your opinion about putting package name in the function name? Can it be just `transform_module()`?

Reply:
I'm a little bit confused what the convention is - also looking into other functions

Reply:
I think it can, but we may want to do this in a different issue and apply the same convention package-wide.

If my memory serves me, weeks/months ago we were all discussing with you @pawelru that `module` and `modules` should have the `teal_` prefix.

Reply:
that's exactly the part that I'm currently struggle with :)

Reply:
Naming convention could be improved, we can discuss it on another issue for sure

Reply:
There's an issue that deals with naming convention: https://github.com/insightsengineering/teal/issues/1035#issuecomment-2507919214